src/pybind/mgr/cephadm/services/services_map.py

```diff
@@ -0,0 +1,28 @@
+# cephadm related services
+from .cephadmservice import CephadmAgent
+from .cephadmservice import CephExporterService
+from .node_proxy import NodeProxy
+from .iscsi import IscsiService
+from .jaeger import JaegerAgentService
+from .mgmt_gateway import MgmtGatewayService
+# monitoring
+from .monitoring import PrometheusService
+from .monitoring import GrafanaService
+from .monitoring import AlertmanagerService
+from .monitoring import PromtailService
+from .monitoring import NodeExporterService
+
+# Define services mapping dictionary
+ServiceMap = {
+    CephadmAgent.TYPE: CephadmAgent,
+    NodeProxy.TYPE: NodeProxy,
+    IscsiService.TYPE: IscsiService,
+    PrometheusService.TYPE: PrometheusService,
+    GrafanaService.TYPE: GrafanaService,
+    AlertmanagerService.TYPE: AlertmanagerService,
+    PromtailService.TYPE: PromtailService,
+    CephExporterService.TYPE: CephExporterService,
+    NodeExporterService.TYPE: NodeExporterService,
+    JaegerAgentService.TYPE: JaegerAgentService,
+    MgmtGatewayService.TYPE: MgmtGatewayService
```

Comment:
Thanks for pointing that out, I wasn't aware of the `self.cephdam_services`. Since it serves exactly the same purpose I'll just go and us it directly. As you said services with no dependencies will return an empty list (from the base class).

Reply:
@phlogistonjohn @adk3798 I like the idea of automatically register/generates the services map (cephadm_services) instead of having to specify the whole list. However, in order to avoid putting more changes on this PR (which already has a lot), and to avoid mix this change with the dependencies movement I think it's better to have a dedicated PR, so I went and created the PR https://github.com/ceph/ceph/pull/61114 on top of this one with a new commit. I'll be rebasin it later once this one is merged.